Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+- Splash screen
+    - PR screenshot: “Your code is bad and you are bad. Have a bad day.”
+- Importance
+    - Who's done code review before?
+        - Who's done it sync?
+            - We don't really do that, aside from maybe some pair
+              programming.
+        - Async?
+    - Pretty much all code at Mozilla gets reviewed before it gets
+      merged.
+        - It's a great way to…
+            - Improve the code
+                - I like to think I'm good at this, but people still
+                  find improvements to make in my stuff almost every
+                  time.
+            - Learn the codebase and the craft
+                - It informs other people what's going on in the
+                  codebase so you're not stuck with the sole
+                  responsibility of maintaining it.
+                - It's as much about teaching as fixing. It's a major
+                  way we bring junior people up the ladder.
+                - Tests may show the code works, but reviews ensure
+                  it's easy to understand.
+    - But open source runs on enthusiasm
+        - Even when you're getting paid to do it
+            - There's a lot of intrinsic motivation around here.
+    - “Crisitunity”
+        - We can drive the desire to contribute out of people, even
+          drive out the people themselves.
+        - Or we can magnify our power through teaching and
+          encouragement.
+        - I will show you how now, brown cow.
+        - For example…
+              …then continue on to goods/bads.
+- Good/bads
+      These will be a series of before/after PR screenshots with
+      discussion.
+    - Clarity
+          “This makes frob() buggy.” vs. “Now if we call frob(), there
+          will be a mem leak as we allocate the whatzit.”
```

Comment:
This is one that annoys me currently. I often get review comments and they're in such shorthand I haven't got a clue what the person is talking about which requires several rounds over several hours of replies.

Sometimes, I get "lecturey" and show Python blocks and links to relevant things. I think seeing examples and seeing code can often bridge the communication gaps when the two people don't know each other really well (and even when they do) such as when a person is new to the project, doesn't have English as a first language, I haven't had my coffee yet this morning, etc.

I mention it because maybe it helps to add a sentence like:

> Example code snippets, REPL sessions, links to relevant documentation, etc.